src/node-lib/curriculum-api-2023/queries/unitListing/filters/getAllCategroies.ts

```diff
@@ -0,0 +1,47 @@
+import { SyntheticUnitvariantLessons } from "@oaknational/oak-curriculum-schema";
+import { kebabCase } from "lodash";
+
+type Category = {
+  label: string;
+  iconName: string;
+  slug: string;
+};
+
+// category icons don't have a consistent naming convention
+const categoryIconMap = {
+  Grammar: "subject-english-grammar",
+  Handwriting: "subject-english-handwriting",
+  "Reading, writing & oracy": "subject-english-reading-writing-oracy",
+  Spelling: "subject-english-spelling",
+  Vocabulary: "subject-english-vocabulary",
+  Physics: "subject-physics",
+  Biology: "subject-biology",
+  Chemistry: "subject-chemistry",
+};
+export type CategoryKeys = keyof typeof categoryIconMap;
+
+export const getAllCategories = (
+  parsedRawUnits: SyntheticUnitvariantLessons[],
+): Category[] => {
+  return Array.from(
+    parsedRawUnits
+      .reduce((acc, unit) => {
+        if (!unit.unit_data.subjectcategories) {
+          return acc;
+        }
+        (unit.unit_data.subjectcategories as CategoryKeys[]).forEach(
+          (category) => {
+            if (typeof category === "string" && !acc.has(category)) {
```

Comment:
defining the category keys as a zod schema and then parsing might be a bit neater here? doesn't need to be done immediately but it would also be good if category keys then live in curiculum-schema lib

Reply:
Yeah I think we need a way for this to generate the category names, maybe a bigger question about subject-icon names too